common/interfaces/desktop_app_interface/usb_api_priv.h

```diff
@@ -16,22 +16,189 @@
+                             ///< PKT_TYPE_STATUS_ACK & PKT_TYPE_ERROR.
+  uint8_t packet_type;       ///< Defines the packet type. @see comm_packet_type
+  uint8_t payload_length;    ///< Length of the payload in the current packet
+  uint32_t timestamp;        /** Packet preparation timestamp of host machine in
+                                milliseconds. Each entity (host & peripheral) will
+                                reset to 0 across resets. */
+} comm_header_t;
+
+/**
+ * @brief struct for message received from desktop
+ * @details This is the application buffer for the message received from the
+ * desktop application. The entire payload is aggregated into this buffer before
+ * handing it over to the application. Similarly, the response from the
+ * application is stored into this buffer before sending it to the desktop
+ * application. The aggregation and fragmentation is handled by the
+ * communication layer. For members of the raw serialized message, refer
+ * comm_raw_message_t.
+ *
+ * @see packet_t, comm_raw_payload_t
+ * @since v1.0.0
+ *
+ * @note
+ */
+typedef struct comm_payload {
+  uint16_t proto_data_length;    ///< Length of the protobuf serialized data in
+                                 ///< the cmd payload
+  uint16_t raw_data_length;    ///< Length of the raw serialization data in the
+                               ///< cmd payload
+  uint8_t *proto_data;    ///< Protobuf serialization data in the cmd payload
+  uint8_t *raw_data;      ///< Raw serialization data in the cmd payload
+} comm_payload_t;
+
+/**
+ * @brief Single package struct
+ * @details
+ *
+ * @see comm_header_t
+ * @since v1.0.0
+ *
+ * @note
+ */
+typedef struct packet {
+  comm_header_t header;
+  const uint8_t *payload;
+  comm_libusb__interface_e interface;
+} packet_t;
+
+typedef struct comm_status {
+  // Application info
+  uint8_t app_busy_status;
+  uint8_t abort_disabled;
+  uint8_t curr_flow_status;
+
+  // Peripheral info
+  uint16_t curr_cmd_seq_no;
+  uint8_t curr_cmd_state;
+
+  // Details of the command receive in progress (not to be sent to host)
+  uint16_t curr_cmd_chunk_no;
+  uint16_t curr_cmd_received_length;
+
+  // Host sync status (not to be sent to host)
+  uint32_t host_sync_time;
+  uint8_t host_sync_fails;
+} comm_status_t;
+
 /*****************************************************************************
  * EXPORTED VARIABLES
  *****************************************************************************/
 
+// received/ready-to-send cmd can be decoded/encoded into this buffer
+extern uint8_t comm_io_buffer[COMM_BUFFER_SIZE];
+extern comm_payload_t comm_payload;
+extern comm_status_t comm_status;
+
 /*****************************************************************************
  * GLOBAL FUNCTION PROTOTYPES
  *****************************************************************************/
+
+static inline void comm_set_payload_struct(uint16_t proto_len,
```

Comment:
ATCG: Inline functions are not preferred as they are harder to mock for unit tests. Suggest moving the function definition to source file.

Reply:
Resolving as not introduced in this pr and defined in _priv file